src/nbl/video/utilities/IUtilities.cpp

```diff
@@ -1,171 +1,75 @@
 #include "nbl/video/utilities/IUtilities.h"
+#include "nbl/asset/filters/CConvertFormatImageFilter.h"
 #include <numeric>
 
 namespace nbl::video
 {
-
 void IUtilities::updateImageViaStagingBuffer(
     IGPUCommandBuffer* cmdbuf, IGPUFence* fence, IGPUQueue* queue,
-    asset::ICPUBuffer const* srcBuffer, const core::SRange<const asset::IImage::SBufferCopy>& regions, video::IGPUImage* dstImage, asset::IImage::E_LAYOUT dstImageLayout,
+    asset::ICPUBuffer const* srcBuffer, asset::E_FORMAT srcFormat, video::IGPUImage* dstImage, asset::IImage::E_LAYOUT dstImageLayout, const core::SRange<const asset::IImage::SBufferCopy>& regions,
     uint32_t& waitSemaphoreCount, IGPUSemaphore*const * &semaphoresToWaitBeforeOverwrite, const asset::E_PIPELINE_STAGE_FLAGS* &stagesToWaitForPerSemaphore)
 {
     const auto& limits = m_device->getPhysicalDevice()->getLimits();
     const uint32_t allocationAlignment = static_cast<uint32_t>(limits.nonCoherentAtomSize);
 
     auto* cmdpool = cmdbuf->getPool();
+    
+    if(cmdbuf == nullptr || fence == nullptr || queue == nullptr || regions.size() == 0 || dstImage == nullptr || (srcBuffer == nullptr || srcBuffer->getPointer() == nullptr))
+    {
+        assert(false);
+        return;
+    }
```

Comment:
empty regions isnt something to `assert(false)` over

Reply:
why use copy when you don't want to copy anything?

Reply:
ok nvm, I'll make it not assert on empty region, nothing too important